llama_stack/distribution/server/auth_providers.py

```diff
@@ -162,18 +177,99 @@ async def close(self):
             self._client = None
 
 
+JWT_AUDIENCE = "llama-stack"
+
+
+class JWTAuthProviderConfig(BaseModel):
+    # The JWKS URI for collecting public keys
+    jwks_uri: str
+    algorithm: str = "RS256"
+    cache_ttl: int = 3600
+
+
+class JWTAuthProvider(AuthProvider):
+    """
+    JWT token authentication provider that validates a JWT token and extracts access attributes.
+
+    This should be the standard authentication provider for most use cases.
+    """
+
+    def __init__(self, config: JWTAuthProviderConfig):
+        self.config = config
+        self._jwks_at: float = 0.0
+        self._jwks: dict[str, str] = {}
+
+    async def validate_token(self, token: str, scope: dict | None = None) -> TokenValidationResult:
+        """Validate a token using the JWT token."""
+        await self._refresh_jwks()
+
+        try:
+            kid = jwt.get_unverified_header(token)["kid"]
+            if kid not in self._jwks:
+                raise ValueError(f"Unknown key ID: {kid}")
+            key = self._jwks[kid]
+            claims = jwt.decode(
+                token,
+                key,
+                algorithms=[self.config.algorithm],
+                audience=JWT_AUDIENCE,
+                options={"verify_exp": True},
+            )
+        except Exception as exc:
+            raise ValueError(f"Invalid JWT token: {token}") from exc
+
+        # There are other standard claims, the most relevant of which is `scope`.
+        # We should incorporate these into the access attributes.
+        principal = f"{claims['iss']}:{claims['sub']}"
```

Comment:
Question (of minor importance): Is qualification by iss necessary? (Do you anticipate having separate issuers with a shared endpoint for keys?) The fully qualified value would be a little more awkward which is of course justified if multiple issuers are anticipated.

Reply:
Yeah good point, I don't think the qualification is necessary.

Reply:
OK I looked at #2187 now. It does seem like an alternate way to verify the token and seems useful. However, I still wonder whether we could make a single `OAuth2AuthProvider` here which could have different behavior depending on configuration. Then the claims processing will be unified and I would really want a _SINGLE_ auth provider in the Stack if at all possible.

Reply:
@grs sounds good. I will rename my provider appropriately, steal a bit of your claims processing logic and add some tests.